src/session/session_compact.c

```diff
@@ -361,6 +360,11 @@ __compact_worker(WT_SESSION_IMPL *session)
                   "conflicting checkpoint. Compaction of this data handle will be retried.",
                   session->op_handle[i]->name);
             }
+
+            /* Compaction was interrupted internally. */
+            else if (ret == EINTR)
+                ret = 0;
```

Comment:
I am wondering here if we should `another_pass` to `false`.
If a previous dhandle was successfully compacted, we want to get to the checkpoints to reclaim the space for it. On the other hand, we may process again that same dhandle where there is no work to do.

Reply:
Before the compact operation `another_pass` is set to `false`, so there is no need to set it again.

Reply:
As discussed on Slack, `another_pass` can be set to `true` because of a previous dhandle that was successfully compcted. We need to keep that flag as we need to trigger the checkpoint to reclaim the space instead of losing the work we have done.